`ArgminWeightedDot` error on `main` branch

Hi, I tried to use the library using the latest commit with a git dependency, but encountered the following error (and many other similar ones):
```rust
   Checking argmin-math v0.1.0 (https://github.com/argmin-rs/argmin?rev=97d89c2d3634f84eb0f3957a20cb96d040161469#97d89c2d)
    Checking argmin v0.5.0 (https://github.com/argmin-rs/argmin?rev=97d89c2d3634f84eb0f3957a20cb96d040161469#97d89c2d)
error[E0599]: no method named `weighted_dot` found for type parameter `P` in the current scope
   --> ...\.cargo\git\checkouts\argmin-a97cfb01ebe8501e\97d89c2\argmin\src\solver\quasinewton\sr1_trustregion.rs:298:26
    |
298 |         let tmp2: F = sk.weighted_dot(&hessian, &sk);
    |                          ^^^^^^^^^^^^ method not found in `P`
    |
    = help: items from traits can only be used if the type parameter is bounded by the trait
help: the following trait defines an item `weighted_dot`, perhaps you need to restrict type parameter `P` with it:
    |
191 |         + ArgminZeroLike + ArgminWeightedDot,
    |                          +++++++++++++++++++
```

This was really confusing, as it's just the library code, so I investigated a little.
The trait is implemented as a blanket impl, which should be visible with just the default `primitives` feature, I have no idea why it isn't.

Turns out that enabling the `ndarrayl` feature, which I would need anyway, solved the issue somehow.
The CI and most building instructions I saw include choosing a feature for the backend, or using `--all-features`, so that's what most people will probably do, but the error if you forget/don't see it is extraordinarily confusing, so I wanted to leave this note.

If *at least* one backend must be on, maybe leave a `cfg`ed `compile_error!` that explains this situation when building.

--- a/argmin/Cargo.toml
+++ b/argmin/Cargo.toml
@@ -22,7 +22,7 @@ num-traits = { version = "0.2" }
 rand = { version = "0.8.3" }
 rand_xoshiro = { version = "0.6.0", features = ["serde1"] }
 thiserror = "1.0"
-argmin-math = { path = "../argmin-math", default-features = false }
+argmin-math = { path = "../argmin-math", default-features = false, features = ["primitives"] }
 # optional
 bincode = { version = "1.3.3", optional = true }
 ctrlc = { version = "3.1.2", optional = true }

